pkg/networkservice/common/prioritymechanisms/client.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2022 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package prioritymechanisms prioritize mechanisms according to the list
+package prioritymechanisms
+
+import (
+	"context"
+	"sort"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+type priorityMechanismsClient struct {
```

Comment:
How do we plan to use this in forwarder-vpp app?

Reply:
We add it to the forwarder's chain, for example after `filtermechanisms` - https://github.com/networkservicemesh/sdk-vpp/blob/main/pkg/networkservice/chains/forwarder/server.go#L152
To pass values, we will use `NSM_MECHANISM_PRIORITY` env that is a prioritized list (`[]string`) - the first value has the highest priority.

For example, if we have
```
            - name: NSM_MECHANISM_PRIORITY
              value: ipsec
```
this means that the mechanismPreferences order will be: `{ IPSec, Wireguard }`

Reply:
Looks good to me.

/cc @edwarnicke